packages/rewriter/lib/js.ts

```diff
@@ -0,0 +1,72 @@
+import * as astTypes from 'ast-types'
+import Debug from 'debug'
+import { jsRules } from './js-rules'
+import * as recast from 'recast'
+import * as sourceMaps from './util/source-maps'
+
+const debug = Debug('cypress:rewriter:js')
+
+const defaultPrintOpts: recast.Options = {
+  // will only affect reprinted quotes
+  quote: 'single',
+}
+
+type OriginalSourceInfo = { url: string, js: string }
+
+// a function that, given source info, returns an id that can be used to build the sourcemap later
+export type DeferSourceMapRewriteFn = (sourceInfo: OriginalSourceInfo) => string
+
+export function rewriteJsSourceMap (url: string, js: string, inputSourceMap: any): any {
+  try {
+    const { sourceFileName, sourceMapName, sourceRoot } = sourceMaps.getPaths(url)
+
+    const ast = recast.parse(js, { sourceFileName })
+
+    astTypes.visit(ast, jsRules)
+
+    return recast.print(ast, {
+      inputSourceMap,
+      sourceMapName,
+      sourceRoot,
+      ...defaultPrintOpts,
+    }).map
+  } catch (err) {
+    debug('error while parsing JS %o', { err, js: js.slice ? js.slice(0, 500) : js })
+
+    return { err }
+  }
+}
+
+export function _rewriteJsUnsafe (url: string, js: string, deferSourceMapRewrite?: DeferSourceMapRewriteFn): string {
+  const ast = recast.parse(js)
+
+  astTypes.visit(ast, jsRules)
```

Comment:
catch errors here and return JS that signals Cypress that an error rewriting the AST has occurred